### PR TITLE
fix(compiler): capture data bindings for content projection purposes in blocks

### LINKED ISSUE #54872: Dynamically generated content is not projected
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

It looks like that the new template pipeline which is enabled by default since version 17.3.0 does not properly handle projected contant which is created dynamically.

As you can see in the stackblitz example, if the element is statically added as a child of the container element, it is properly projected. However, if the content is dynamically created (e.g. with an `if` or `for`), then the container will be empty.

It seems to be a rendering/DOM issue only, because the children are properly queried (see the console output).

Might be related: #54840

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-nnuszu?file=src%2Fmain.ts

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.3.0
Node: 18.18.0
Package Manager: npm 10.2.3
OS: linux x64

Angular: 17.3.0
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1703.0
@angular-devkit/build-angular   17.3.0
@angular-devkit/core            17.3.0
@angular-devkit/schematics      17.3.0
@schematics/angular             17.3.0
rxjs                            7.8.1
typescript                      5.4.2
zone.js                         0.14.4
```

### Anything else?

The `ng version` is printed by the stackblitz example, we have the exact same versions except the OS which is not related in this case.

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes a regression in the template pipeline where data bindings weren't being captured for content projection purposes.

Fixes #54872.

### CODE REVIEW

File: packages/compiler/src/template/pipeline/src/ingest.ts [hunk 1]
```diff
@@ -359,7 +359,7 @@ function ingestIfBlock(unit: ViewCompilationUnit, ifBlock: t.IfBlock): void {
       cView.contextVariables.set(ifCase.expressionAlias.name, ir.CTX_REF);
     }
 
-    let ifCaseI18nMeta = undefined;
+    let ifCaseI18nMeta: i18n.BlockPlaceholder|undefined = undefined;
```
Comment (crisbeto) [inline packages/compiler/src/template/pipeline/src/ingest.ts:362]: Unrelated change, I just saw that vscode was flagging this as an error.

File: packages/compiler/src/template/pipeline/src/ingest.ts [hunk 2]
```diff
@@ -1269,18 +1269,28 @@ function ingestControlFlowInsertionPoint(
...
   if (root !== null) {
+    // Collect the static attributes for content projection purposes.
     for (const attr of root.attributes) {
       const securityContext = domSchema.securityContext(NG_TEMPLATE_TAG_NAME, attr.name, true);
       unit.update.push(ir.createBindingOp(
           xref, ir.BindingKind.Attribute, attr.name, o.literal(attr.value), null, securityContext,
           true, false, null, asMessage(attr.i18n), attr.sourceSpan));
     }
 
+    // Also collect the inputs since they participate in content projection as well.
```
Comment (crisbeto) [inline packages/compiler/src/template/pipeline/src/ingest.ts:1283]: For reference, [this](https://github.com/angular/angular/blob/17.2.x/packages/compiler/src/render3/view/template.ts#L1555) is where the TDB was constructing the attributes array and [this](https://github.com/angular/angular/blob/17.2.x/packages/compiler/src/render3/view/template.ts#L2042) is the code that was capturing both the static attributes and the inputs.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: fw-core

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-compiler

**Comment by dylhunn** [comment]
(thanks for fixing this issue @crisbeto!)

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 879bd80b571368bdc6e65c5415d98ad0f546267a.
